src/components/modals/ManageGroupsAndTypesModal.svelte

```diff
@@ -0,0 +1,816 @@
+              content: 'Delete External Source',
+              placement: 'bottom',
+            },
+            hasDeletePermission: hasDeletePermission,
+            rowData: params.data,
+          },
+          target: actionsDiv,
+        });
+
+        return actionsDiv;
+      },
+      cellRendererParams: {
+        deleteExternalEventType,
+      } as CellRendererParams,
+      headerName: '',
+      resizable: false,
+      sortable: false,
+      suppressAutoSize: true,
+      suppressSizeToFit: true,
+      width: 25,
+    },
+    {
+      cellClass: 'action-cell-container',
+      cellRenderer: (params: ModalCellRendererParams) => {
+        const actionsDiv = document.createElement('div');
+        actionsDiv.className = 'actions-cell';
+        new DataGridActions({
+          props: {
+            rowData: params.data,
+            viewCallback: params.viewExternalEventType,
+            viewTooltip: {
+              content: 'View External Event Type',
+              placement: 'bottom',
+            },
+          },
+          target: actionsDiv,
+        });
+        return actionsDiv;
+      },
+      cellRendererParams: {
+        viewExternalEventType,
+      } as CellRendererParams,
+      headerName: '',
+      resizable: false,
+      sortable: false,
+      suppressAutoSize: true,
+      suppressSizeToFit: true,
+      width: 40,
+    },
+  ];
+
+  const modalColumnSizeNoDetail: string = '1fr 3px 0fr';
+  const modalColumnSizeWithDetailDG: string = '3fr 3px 1.3fr';
+  const modalColumnSizeWithDetailEET: string = '2fr 3px 1.2fr';
+  const modalColumnSizeWithDetailEST: string = '2fr 3px 1.2fr';
+  let modalColumnSize: string = modalColumnSizeNoDetail;
+
+  let newTypeName: string = "";
+  let newTypeSourceType: string = "";
+  let newTypeError: string | null = null;
+
+  let selectedDerivationGroup: DerivationGroup | undefined = undefined;
+  let selectedDerivationGroupSources: ExternalSourceSlim[] = [];
+
+  let selectedExternalSourceType: ExternalSourceType | undefined = undefined;
+  let selectedExternalSourceTypeDerivationGroups: DerivationGroup[] = [];
+
+  let selectedExternalEventType: ExternalEventType | undefined = undefined;
+  let selectedExternalEventTypeDerivationGroups: DerivationGroup[] = [];
+
+  $: selectedDerivationGroupSources = $externalSources.filter(
+    source => selectedDerivationGroup?.name === source.pkey.derivation_group_name,
+  );
+
+  $: selectedExternalSourceTypeDerivationGroups = $derivationGroups.filter(dg => {
+    if (selectedExternalSourceType !== undefined) {
+      return dg.source_type_name === selectedExternalSourceType.name;
+    } else {
+      return false;
+    }
+  });
+
+  $: selectedExternalEventTypeDerivationGroups = $derivationGroups.filter(dg => {
+    if (selectedExternalEventType !== undefined) {
+      return dg.event_types.includes(selectedExternalEventType.name);
+    } else {
+      return false;
+    }
+  });
+
+  async function onCreateType() {
```

Comment:
Seeing successful derivation group creation but not seeing the new derivation group show up in the modal after or when I refresh the page. Not seeing any errors in backend logs. Any idea if I'm doing something wrong or if this is expected?

Reply:
Think this was a database issue and you might just be out-of-date by a commit or two - try re-building and let us know if you're still seeing issues, that table should auto-populate with your new derivation group/external event type/external source type

Reply:
Can confirm that with the current version of the database and UI, this issue is not present.